### DISCUSSION TITLE
Navigation menu issue

### DISCUSSION THREAD

**Discussion by ZioZenigata** [serenity-is/Serenity#6466] [Help]
Hi,
I've a problem with navigation menu.
This is my controller

[Route("TS")]
[Route("TS/{fn?}/{id?}")]
[PageAuthorize(typeof(UserForm.UserFormRow))]
public ActionResult TS(string fn, int? id = null)
{
   return View(MVC.Views.TS.UserForm.UserFormIndex, string.Format("{0}/{1}", fn, id)); 
}

and this is my Navigation.cs:

[assembly: NavigationLink(int.MaxValue, "TS/UserForm", typeof(MyPages.MyController), action: "TS", icon: null)]

This link https://localhost:55859/TS open page and show link (TS/UserForm) in sidebar correctly

This link https://localhost:55859/TS/RO/4 open page correctly but and don't show link (TS/UserForm) in sidebar

**I need two different actions and see the same node in sidebar menu**

How can I solve my issue?
Thanks in advance

**Comment by stixoffire** [comment]
Please show some pictures of what you are seeing, hi-lite what is wrong - as I am trying to under stand what you are describing.

While I think I am following - that you want two links in the nav tree one for each web link .

[assembly: NavigationLink(int.MaxValue, "TS/UserForm", typeof(MyPages.MyController), action: "TS", icon: null)]
[assembly: NavigationLink(int.MaxValue, "TS/RO/4", typeof(MyPages.MyController), action: "TS", icon: null)]

as you can see I am not understanding what you desire - pictures will help better. With markup this is right , this is wrong .. result should look like this ..

**Comment by ZioZenigata** [comment]
Hi Ken,
Sorry I'm late but I was on vacation. Victor has already answered me with
the email I am attaching to you. This "not too elegant" way solved the
problem
Thank you

Il giorno ven 12 ago 2022 alle ore 23:33 Ken ***@***.***> ha
scritto:
